### ISSUE TITLE
seed is not honored for train/validation split or minibatch order when theta/x are precomputed

### ISSUE THREAD

**Issue by pedroliman** [pedroliman/neuralsbi#213]
## Bug

`npe()`, `nle()`, and `nre()` document `seed` as "Optional integer seed for reproducibility" (`R/npe.R:68`), but the seed only reaches R's base RNG through `simulate_for_sbi()` (`R/npe.R:341`, `set.seed(seed)`), which is only called on the branch of `prepare_simulations()` (`R/npe.R:196-225`) that runs the simulator itself. When a caller supplies pre-computed `theta`/`x` directly (a documented, first-class code path, see `@param theta,x`), that branch is skipped entirely, and `set.seed(seed)` is never called on the base R RNG.

Training itself then depends on R's base RNG in two places in `train_restarts()` (`R/train.R`):

- the train/validation split, `perm <- sample.int(n)` (`R/train.R:146`)
- every epoch's minibatch shuffle, `sample.int(n_tr)` inside the training loop (`R/train.R:189`)

`train_restarts()` only seeds torch's RNG (`torch::torch_manual_seed(seed)`, `R/train.R:139`), not R's. So with pre-computed `theta`/`x`, the split and minibatch order are drawn from whatever the ambient R RNG state happens to be at call time.

## Impact

Two calls with identical `seed`, identical pre-computed `theta`/`x`, and no simulator can produce different trained MDN/MAF/NSF/NRE models, because the train/val split and epoch batch order differ. This contradicts the documented contract and is inconsistent with the rest of the package -- `npe_sequential()` (`R/sequential.R:103`), `c2st()` (`R/c2st.R:93`), and the NLE MCMC posterior (`R/nle_posterior.R:208`) all call `set.seed(seed)` directly for exactly this reason.

The existing reproducibility test (`tests/testthat/test-nle.R:263-271`, "pre-computed simulations give the same fit as running the simulator") doesn't catch this: it uses `density_estimator = "linear_gaussian"`, a closed-form fit with no randomness in training at all, so it never exercises `train_restarts()`'s RNG dependence.

## Suggested fix

Seed R's base RNG once, unconditionally of which `prepare_simulations()` branch runs -- e.g. `if (!is.null(seed)) set.seed(seed)` near the top of the exported `npe()`/`nle()`/`nre()` entry points (matching the pattern already used in `sequential.R`, `c2st.R`, and `nle_posterior.R`), so the pre-computed-`theta`/`x` path gets the same guarantee the simulator path already has via `simulate_for_sbi()`.

## Repro sketch

```r
theta <- matrix(rnorm(2000), ncol = 2)
x <- theta + matrix(rnorm(2000, sd = 0.1), ncol = 2)
fit1 <- npe(theta = theta, x = x, density_estimator = "mdn", seed = 1)
fit2 <- npe(theta = theta, x = x, density_estimator = "mdn", seed = 1)
# fit1 and fit2's trained weights (and val loss history) can differ,
# because the train/val split and minibatch order aren't seeded.
```

Found during a routine code-quality audit; not yet reproduced end-to-end against the actual torch training loop (this environment doesn't have libtorch installed), but the RNG-dependence is confirmed by direct code inspection of `R/train.R` and `R/npe.R`.
